### PR TITLE
release: prepare for 0.4.1

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,12 +5,14 @@ All notable changes to this project will be documented in this file.
...
 
-## [Unreleased]
+## [0.4.1] - 2023-12-21
 
 #### Changed
  - The point multiplication algorithm used for ECDH operations (module `ecdh`) was replaced with a slightly faster one.
  - Optional handwritten x86_64 assembly for field operations was removed because modern C compilers are able to output more efficient assembly. This change results in a significant speedup of some library functions when handwritten x86_64 assembly is enabled (`--with-asm=x86_64` in GNU Autotools, `-DSECP256K1_ASM=x86_64` in CMake), which is the default on x86_64. Benchmarks with GCC 10.5.0 show a 10% speedup for `secp256k1_ecdsa_verify` and `secp256k1_schnorrsig_verify`.
 
+#### ABI Compatibility
+This release is compatible with the ABI of 0.4.0 and 0.3.x.
```
Comment (real-or-random) [inline CHANGELOG.md:15]: nit: Use same wording as below?  


```suggestion
The ABI is compatible with versions 0.4.0 and 0.3.x.
```

**Review by real-or-random — APPROVED** [review]
ACK d60270bbbbf9306fa650adfce12e105b117813c2

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,12 +5,14 @@ All notable changes to this project will be documented in this file.
...
 
-## [Unreleased]
+## [0.4.1] - 2023-12-21
 
 #### Changed
  - The point multiplication algorithm used for ECDH operations (module `ecdh`) was replaced with a slightly faster one.
  - Optional handwritten x86_64 assembly for field operations was removed because modern C compilers are able to output more efficient assembly. This change results in a significant speedup of some library functions when handwritten x86_64 assembly is enabled (`--with-asm=x86_64` in GNU Autotools, `-DSECP256K1_ASM=x86_64` in CMake), which is the default on x86_64. Benchmarks with GCC 10.5.0 show a 10% speedup for `secp256k1_ecdsa_verify` and `secp256k1_schnorrsig_verify`.
 
+#### ABI Compatibility
+The ABI is compatible with versions 0.4.0 and 0.3.x.
```
[the remark this answers is not in the document]
Comment (jonasnick) [reply CHANGELOG.md:15]: added the word "backward"

**Review by real-or-random — APPROVED** [review]
ACK 672053d8015fe8ac8d411ef79fbcd97abc045b56

**Review by hebasto — APPROVED** [review]
ACK 672053d8015fe8ac8d411ef79fbcd97abc045b56

**Pull request merged**